src/libs/actions/User.ts

```diff
@@ -12,84 +11,93 @@ import * as ReportActionsUtils from '@libs/ReportActionsUtils';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {FrequentlyUsedEmoji} from '@src/types/onyx';
+import type Login from '@src/types/onyx/Login';
+import {OnyxServerUpdate} from '@src/types/onyx/OnyxUpdatesFromServer';
+import type OnyxPersonalDetails from '@src/types/onyx/PersonalDetails';
+import ReportAction from '@src/types/onyx/ReportAction';
 import * as Link from './Link';
 import * as OnyxUpdates from './OnyxUpdates';
 import * as PersonalDetails from './PersonalDetails';
 import * as Report from './Report';
 import * as Session from './Session';
 import redirectToSignIn from './SignInRedirect';
 
-let currentUserAccountID = '';
+type CustomStatus = {text: string; emojiCode: string; clearAfter?: string};
+type BlockedFromConciergeNVP = {expiresAt: number};
+
+let currentUserAccountID = -1;
 let currentEmail = '';
 Onyx.connect({
     key: ONYXKEYS.SESSION,
-    callback: (val) => {
-        currentUserAccountID = lodashGet(val, 'accountID', -1);
-        currentEmail = lodashGet(val, 'email', '');
+    callback: (value) => {
+        currentUserAccountID = value?.accountID ?? -1;
+        currentEmail = value?.email ?? '';
     },
 });
 
-let myPersonalDetails = {};
+let myPersonalDetails: Partial<OnyxPersonalDetails> = {};
```

Comment:
Why do you use `Partial` here?

Reply:
@fvlvte maybe we can try to go with `OnyxPersonalDetails | Record<string, never>;` then?